tests/everest/test_config_validation.py

```diff
@@ -506,6 +509,18 @@ def test_that_install_data_allows_runpath_root_as_target(
             {"source": "baz/", "link": True, "target": 3},
             "Input should be a valid string",
         ),
+        (
+            {"link": True, "target": "bar.json"},
+            "Either source or data must be provided",
+        ),
+        (
+            {"source": "baz/", "data": {"foo": 1}, "link": True, "target": "bar.json"},
+            "The data and source options are mutually exclusive",
+        ),
+        (
+            {"data": {"foo": 1}, "link": True, "target": "bar.txt"},
+            "Invalid target extension",
```

Comment:
(hmm, codecov says this is never hit, something does not work somewhere..)

Reply:
`pytest` runs that test without issue, so maybe codecov has issues with parametrized tests?